File: 백준/Silver/2961. 도영이가 만든 맛있는 음식/도영이가 만든 맛있는 음식.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[][] tasty = new int[n][2];
      
        for(int i = 0;i<n;i++){
            StringTokenizer st = new StringTokenizer(br.readLine());
            tasty[i][0] = Integer.parseInt(st.nextToken());
            tasty[i][1] = Integer.parseInt(st.nextToken());
        }
        int minDiff = Integer.MAX_VALUE;

        for(int i = 1;i<(1<<n);i++){
            int sour = 1;
            int bitter = 0;

            for(int j = 0;j<n;j++){
                if((i&(1<<j))!=0){
                    sour *= tasty[j][0];
                    bitter += tasty[j][1];
                }
            }
            minDiff = Math.min(minDiff,Math.abs(sour-bitter));
        }
      
    System.out.println(minDiff);
    }
    
}
    

File: 백준/Silver/2961. 도영이가 만든 맛있는 음식/도영이가 만든 맛있는 음식.py
import sys
input = sys.stdin.readline

n = int(input())
tasty = [list(map(int, input().split())) for _ in range(n)]

min_diff = float('inf') #최소 차이를 저장할 변수

for i in range(1,1<<n):
    sour = 1
    bitter = 0

    for j in range(n):
        if i&(1<<j):
            sour *= tasty[j][0]
            bitter += tasty[j][1]
    min_diff = min(min_diff, abs(sour-bitter))
print(min_diff)
